enhancements/single-node-production-edge-cluster-profile.md

```diff
@@ -0,0 +1,380 @@
+automatically rather than one at a time. Therefore the console will
+not be deployed by this profile.
+
+The profile describes single-node, all-in-one, deployments, so there
+is no need to support provisioning additional workers. The
+machine-api-operator and cluster-baremetal-operator will not be
+included in clusters using this profile. Remediation (by rebooting or
+reprovisioning the host) will be handled by an orchestration tool 
+running outside of the node.
+
+In the telco RAN use case, high-availability is typically achieved by
+having multiple sites provide coverage to overlapping geographic
+areas. Therefore, use of cluster-based high-availability features will
+be limited (for example, by running a single API service).
+
+Failures in edge deployments are frequently resolved by re-imaging or
+physically replacing the entire host. Combining this fact with the
+previous observation about the approach to providing highly-available
+services lets us draw the conclusion that in-place upgrades do not
+need to be supported by the first iteration of this cluster profile.
+
+The cluster profile will be sufficiently generic to cater to a variety
+of edge computing use cases. As such, OpenShift's usual cluster
+configuration mechanisms will be favored where there is a likelihood
+that there will be edge computing use case with differing requirements.
+For example, there is no expectation that this new profile will use
+the real-time kernel by default - this will continue to be a 
+`MachineConfig` choice as per
+[enhancements/support-for-realtime-kernel](https://github.com/openshift/enhancements/blob/master/enhancements/support-for-realtime-kernel.md). *See open questions*
+
+### User Stories
+
+#### As a user, I can deploy OpenShift in a supported single-node configuration
+
+A user will be able to run the OpenShift installer to create a single-node
+deployment, with some limitations (see non-goals above). The user
+will not require special support exceptions to receive technical assistance
+for the features supported by the configuration.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Some OpenShift components (such as Etcd and Ingress) require
+a minimum of 2 or 3 nodes. The `single-node-production-edge`
+cluster profile will configure these components as appropriate
+for a single node.
+
+Known gaps:
+- `cluster-etcd-operator` will not deploy the etcd cluster without minimum of 3 master nodes (can be changed by enabling `useUnsupportedUnsafeNonHANonProductionUnstableEtcd`)
+- Even with the unsupported feature flag, `etcd-quorum-guard` still requires 3 nodes due to its replica count.
+- `cluster-authentication-operator` will not deploy `OAuthServer` without minimum of 3 master nodes (can be change by enabling `useUnsupportedUnsafeNonHANonProductionUnstableOAuthServer`)
+- `cluster-ingress-operator` deploys the router with 2 replicas. On a single node one will fail to start and the ingress will show as degraded.
```

Comment:
Do we really need ingress in RAN use case?

Reply:
auth currently depends on functional ingress.